### PR TITLE
Create a tiny app for writing the documentation jsons

### PR DESCRIPTION
**Pull request by ylvaselling**
Add simple app for writing the documentation (only asset components and lua scripting).

This will be used by the Jenkins machine to automatically create the .json files for the documentation. It only works on windows and Linux.

To test:
* Use Cmake to generate a project where the docs writer is selected (by default it isn't)
* Set it as startup project in visual studio
* Run, and check the documentation folder in OpenSpace. It should have printed out two .json files, `scriptingApi.json` and `assetComponets.json`. 
* To test it super properly, place these json files in the OpenSpace-Docs/json folder, and then run `make html` (after activating the python virtual environment with `.\.venv\Scripts\activate`). This will generate the new documentation. Open it from the index file in the build folder. 
* See that it looks ok and that the console didn't print out error messages

Done!

### CODE REVIEW

File: apps/DocsWriter/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
+##########################################################################################
+#                                                                                        #
+# OpenSpace                                                                              #
+#                                                                                        #
+# Copyright (c) 2014-2022                                                                #
```
Comment (WeirdRubberDuck) [inline apps/DocsWriter/CMakeLists.txt:5]:
```suggestion
# Copyright (c) 2014-2024                                                                #
```

File: apps/DocsWriter/main.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+#include <openspace/engine/configuration.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/openspaceengine.h>
+#include <openspace/engine/settings.h>
+#include <openspace/documentation/documentationengine.h>
+
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/ghoul.h>
+#include <ghoul/logging/logmanager.h>
```
Comment (WeirdRubberDuck) [inline apps/DocsWriter/main.cpp:35]:
```suggestion
#include <openspace/engine/configuration.h>
#include <openspace/engine/globals.h>
#include <openspace/engine/openspaceengine.h>
#include <openspace/engine/settings.h>
#include <openspace/documentation/documentationengine.h>
#include <ghoul/filesystem/filesystem.h>
#include <ghoul/ghoul.h>
#include <ghoul/logging/logmanager.h>
#include <iostream>
```

File: apps/DocsWriter/main.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        ghoul::logging::LogLevel::Debug,
+        ghoul::logging::LogManager::ImmediateFlush::Yes
+    );
+
+    ghoul::initialize();
+    global::create();
+
+    // In order to initialize the engine, we need to specify the tokens
+    // We start by registering the path of the executable,
+    // to make it possible to find other files in the same directory.
```
Comment (WeirdRubberDuck) [inline apps/DocsWriter/main.cpp:50]:
```suggestion
    // to make it possible to find other files in the same directory
```
nitpicky :)

File: apps/DocsWriter/main.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        "",
+        glm::ivec2(0)
+    );
+    openspace::global::openSpaceEngine->registerPathTokens();
+
+    // Now that we have the tokens we can initialize the engine
+    global::openSpaceEngine->initialize();
+
+    // Print out the documentation to the documentation folder
+    // @TODO: ylvse-2024-05-02 change this directory when integrating with jenkins?
```
Comment (WeirdRubberDuck) [inline apps/DocsWriter/main.cpp:75]:
```suggestion
    // @TODO (ylvse, 2024-05-02) change this directory when integrating with jenkins?
```

we're a bit inconsistent with how we order dates and names in these todo comments, but this is generally the format

**Review by WeirdRubberDuck — APPROVED** [review]
Tried it out now as well.  This is great :)

File: apps/DocsWriter/main.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <openspace/engine/configuration.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/openspaceengine.h>
+#include <openspace/engine/settings.h>
+#include <openspace/documentation/documentationengine.h>
```
Comment (alexanderbock) [inline apps/DocsWriter/main.cpp:29]:
```suggestion
#include <openspace/documentation/documentationengine.h>
#include <openspace/engine/configuration.h>
#include <openspace/engine/globals.h>
#include <openspace/engine/openspaceengine.h>
#include <openspace/engine/settings.h>
```

File: apps/DocsWriter/main.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/ghoul.h>
+#include <ghoul/logging/logmanager.h>
+#include <iostream>
```
Comment (alexanderbock) [inline apps/DocsWriter/main.cpp:33]:
```suggestion
```

File: apps/DocsWriter/main.cpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        std::filesystem::path(argv[0]).parent_path(),
+        ghoul::filesystem::FileSystem::Override::Yes
+    );
+
+    std::filesystem::path configFile = findConfiguration();
+
+    // Register the base path as the directory where the configuration file lives
+    std::filesystem::path base = configFile.parent_path();
+    constexpr std::string_view BasePathToken = "${BASE}";
+    FileSys.registerPathToken(BasePathToken.data(), base);
```
Comment (alexanderbock) [inline apps/DocsWriter/main.cpp:60]:
```suggestion
    FileSys.registerPathToken("${BASE}", base);
```

Comment (alexanderbock) [reply apps/DocsWriter/main.cpp:60]: Doing the `.data()` path technically works, but might be dangerous as a `string_view` doesn't require there to be a `\0` character at the end of the string.  However the `string` constructor taking a `const char*` requires there to be a `\0` terminating character.

In practice this is working anyway, but would be better not to rely on it

File: include/openspace/documentation/documentationengine.h [hunk 8]
```diff
@@ -88,7 +88,7 @@ class DocumentationEngine {
     static DocumentationEngine& ref();
 
     void writeJavascriptDocumentation() const;
-    void writeJsonDocumentation() const;
+    void writeJsonDocumentation(std::string folderName = "${DOCUMENTATION}") const;
```
Comment (alexanderbock) [inline include/openspace/documentation/documentationengine.h:91]: What's the motivation for this change? It doesn't seem to be necessary in this PR?  And using a path token as a default argument seems weird to me?

**Review by alexanderbock — APPROVED** [review]

File: include/openspace/documentation/documentationengine.h (hunk 8 above)
Comment (ylvaselling) [reply include/openspace/documentation/documentationengine.h:91]: Maybe premature optimization, but I was thinking that when we integrate with Jenkins this folder might change, so it could be useful to be able to change it. But I can remove it
